### PR TITLE
Xuejun/deprecated 1.0 part1

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Remove api ov_core_read_model_from_memory()
 - Remove api infer_conv_backprop_output_spatial_shape()
 - *...*

### Tickets:
 - CVS-106449

### CODE REVIEW

File: src/core/include/openvino/op/convolution.hpp [hunk 1]
```diff
@@ -127,28 +127,6 @@ class OPENVINO_API ConvolutionBackpropData : public util::ConvolutionBackPropBas
...
-    ///
-    OPENVINO_DEPRECATED("This member function is deprecated and will be removed soon.")
-    void infer_conv_backprop_output_spatial_shape(const std::vector<Dimension>& input_data_shape,
-                                                  const std::vector<Dimension>& filters_shape,
-                                                  const Strides& strides,
-                                                  const Strides& dilations,
-                                                  const CoordinateDiff& pads_begin,
-                                                  const CoordinateDiff& pads_end,
-                                                  const CoordinateDiff& output_padding,
-                                                  std::vector<Dimension>& output_spatial_shape);
```
Comment (zhaixuejun1993) [inline src/core/include/openvino/op/convolution.hpp:151]: @ilya-lavrenov @vurusovs This API is used in VPUX Plugin, any replacement for this API?
https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/blob/d9a5ea6e1d4888bba55d37e4f8efd9d38365aed6/src/vpux_compiler/src/dialect/IE/ops/convolution_backprop_data.cpp#L97-L105

Comment (vurusovs) [reply src/core/include/openvino/op/convolution.hpp:151]: @praasz @t-jankowski could you take a look?

**Review by vurusovs — APPROVED** [review]

File: src/core/include/openvino/op/convolution.hpp (hunk 1 above)
Comment (praasz) [reply src/core/include/openvino/op/convolution.hpp:151]: Create e.g `GroupConvolutionBackpropData` operator and take its output shape.

**Review by praasz — APPROVED** [review]

**Comment by zhaixuejun1993** [comment]
Blocked by PR: https://github.com/intel-innersource/applications.ai.vpu-accelerators.vpux-plugin/pull/8643

File: src/inference/src/dev/core_impl.cpp [hunk 2]
```diff
@@ -1274,34 +1274,7 @@ void ov::CoreImpl::set_property_for_device(const ov::AnyMap& configMap, const st
...
-                ov::threading::executor_manager()->set_property({{it->first, flag}});
-                config.erase(it);
-            }
-
-            it = config.find(ov::enable_mmap.name());
-            if (it != config.end()) {
-                config.erase(it);
-            }
-        }
+        coreConfig.set_and_update(config);
```
Comment (ilya-lavrenov) [inline src/inference/src/dev/core_impl.cpp:1277]: please, revert this change.
It was supposed to be removed, but currently plugins still need explicitly to know about `cache_dir` property

**Review by t-jankowski — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]
